server/controllers/authController/index.js

```diff
@@ -1,61 +1,88 @@
 import passport from 'passport'
+import dayjs from 'dayjs'
+
 import UserModel from '../../models/UserModel'
 import ConfigModel from '../../models/ConfigModel'
 import RegistrationMailer from '../../mailer/RegistrationMailer'
 import { hash } from '../../utils/crypto/hash'
-import dayjs from 'dayjs'
-import authenticate from '../../strategies/authenticate'
 import { logout } from '../../utils/passport/logout'
+import { isAccountExpired } from '../../utils/passport/helpers'
 
 const AuthController = {
-  authenticate,
-  create: async (req, res, next) => {
+  authenticate: async (req, res, next) => {
     passport.authenticate(
-      'local-signup',
+      'local',
       { session: true },
-      async function (err, existingUser, reqBody) {
-        try {
-          if (err) return res.status(400).json({ error: err.message })
-
-          if (existingUser)
-            return res.status(400).json({ error: 'User has an account' })
-
-          const { appDb } = req.app.locals
-          const password = reqBody.password
-          const uid = reqBody.username
-          const mail = reqBody.mail
-          const display_name = reqBody.fullName
-          const hashedPW = hash(password)
-          const account_expires = dayjs().add(1, 'years').format()
-          const configAttributes = ConfigModel.withDefaults({
-            owner: uid,
-            readers: [uid],
-          })
-          const configuration = await ConfigModel.create(
-            appDb,
-            configAttributes
-          )
-
-          const newUserAttributes = {
-            uid,
-            display_name,
-            password: hashedPW,
-            mail,
-            account_expires,
-            preferences: { config: configuration._id.toString() },
-          }
-          const newUser = await UserModel.create(appDb, newUserAttributes)
-          const registrationMailer = new RegistrationMailer(newUser)
+      async function (err, user, info) {
+        if (err) {
+          return next(err)
+        }
 
-          await registrationMailer.sendMail()
+        const { uid } = user
 
-          return res.status(200).json({ data: newUser })
-        } catch (error) {
-          return res.status(400).json({ error: error.message })
+        const { appDb, dataDb } = req.app.locals
+
+        const userInfo = await UserModel.update(appDb, dataDb, uid, {
+          last_logon: Date.now(),
```

Comment:
Can we use dayjs to set this date with a timezone..?

Reply:
Eh, we could but this is the existing functionality and I'm hesitant to change it.